src/commands/src/helpers/dependency-helpers.ts

```diff
@@ -71,23 +72,23 @@ export async function downloadBuildDependencies(
         throw error;
       }
       commands.push(`mkdir -p ${targetDir}`);
-      await Promise.all(
-        dependencyInfo.dependencies.map(async (dep) => {
-          commands.push(`curl -SfL ${dep.url} -o ${targetDir}/${dep.filename}`);
-          try {
-            return await executeCliCommand({
-              command: 'curl',
-              args: ['--max-time', '10', '-SfL', dep.url, '-o', `${targetDir}/${dep.filename}`],
-              options: options,
-            });
-          } catch (error) {
-            console.error(
-              `ERROR! Could not curl ${dep.url} into ${targetDir}/${dep.filename}. Dependency information: `,
-              dependencyInfo
-            );
-          }
-        })
-      );
+
+      const response = dependencyInfo.dependencies.map((dep) => {
+        const rootDir = targetDir != repoDir ? `${repoDir}/` : '';
+        const curlString = axios
+          .get(dep.url, { timeout: 10000, responseType: 'stream' })
+          .then((res) => {
+            res.data.pipe(fs.createWriteStream(`${rootDir}${targetDir}/${dep.filename}`));
+            return `curl -SfL ${dep.url} -o ${rootDir}${targetDir}/${dep.filename}`;
+          })
+          .catch((error) => {
+            return `ERROR! Could not curl ${dep.url} into ${rootDir}${targetDir}/${dep.filename}. ${error}`;
+          });
+        return curlString;
+      });
+
+      const responseSync = await Promise.all(response);
+      commands = commands.concat(responseSync);
```

Comment:
Such a nit: I think I'd rather see us keep commands as a constant, and use the push method rather than concat which rewrites the entire array each time.